### PR TITLE
backend: Consider the base URL for OIDC's redirect URL

### LINKED ISSUE #238: Headlamp with a base-url and OIDC
**Description**

I'd like to use in cluster deployment with - "-base-url=/headlamp" but when I press the "SIGN-IN" button I get an error - 
**Bad Request
Unregistered redirect_uri ("https://my-host.org/oidc-callback").**

**Impact**

Unreachable callback URL (e.g. **https://YOUR_URL/oidc-callback**) to Dex’s staticClient.redirectURIs

**Environment and steps to reproduce**

1. **Set-up**: 
Kubernetes 1.18.x
Headlamp in cluster deployed with settings:

        - args:
            - '-in-cluster'
            - '-plugins-dir=/headlamp/plugins'
            - '-oidc-client-id=oidc-auth-client'
            - '-base-url=/headlamp'
            - '-oidc-client-secret=xxxx'
            - >-
              -oidc-idp-issuer-url=https://oauth.my-host.org/dex
            - '-oidc-scopes=profile,email,groups'

Ingress:

  name: headlamp-ing
  namespace: headlamp
spec:
  rules:
    - host: my-host.org
      http:
        paths:
          - backend:
              serviceName: headlamp
              servicePort: 80
            path: /headlamp
            pathType: ImplementationSpecific
tls:
    - hosts:
        - my-host.org.tls
      secretName: headlamp-ngx

DEX config:
staticClients:
- id: oidc-auth-client
  name: oidc-auth-client
  redirectURIs:
  - https://my-host.org/headlamp/oidc-callback

4. **Error**: [describe the error that was triggered]

**Expected behavior**

Reachable callback URL (e.g. **https://YOUR_URL/headlamp/oidc-callback)** to Dex’s staticClient.redirectURIs

**Additional information**

Header after pressed "SIGN-IN" button

Request URL: https://oauth.my-host.org/dex/auth?client_id=oidc-auth-client&redirect_uri=https%3A%2F%2Fmy-host.org%2Foidc-callback&response_type=code&scope=openid+profile+email+groups&state=main

### PR DESCRIPTION
**Pull request by joaquimrocha**
If Headlamp is running with a base URL and uses OIDC, the redirect
URL doesn't take into account the base URL which leads to an error.

fixes #238

### CODE REVIEW

**Comment by joaquimrocha** [comment]
This needs to be tested.

File: backend/cmd/headlamp.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func StartHeadlampServer(config *HeadlampConfig) {
 			ClientID:     oidcAuthConfig.ClientID,
 			ClientSecret: oidcAuthConfig.ClientSecret,
 			Endpoint:     provider.Endpoint(),
-			RedirectURL:  fmt.Sprintf("%1s://%2s/oidc-callback", urlScheme, r.Host),
```
Comment (illume) [inline backend/cmd/headlamp.go:310]: :+1:

**Comment by joaquimrocha** [comment]
I have added some more processing to the base URL before joining it so we don't use two `/` in the URL (usually multiple bars in the requests is fine but since we have URL comparisons going on, maybe it's better to ensure that we always get one `/` when separating the URL parts).

**Comment by joaquimrocha** [comment]
Lint's failing, I need to move code out of that huge function. Meeting now, will leave it for tomorrow.

**Comment by joaquimrocha** [comment]
@illume , please take a look. I think that with this change we don't allow regressions so we can safely merge.

**Review by illume — APPROVED** [review]
:+1: thanks

**Pull request merged**